app/src/main/java/ru/kryu/playlistmaker/favourite/ui/view_model/FavouritesViewModel.kt

```diff
@@ -2,14 +2,42 @@ package ru.kryu.playlistmaker.favourite.ui.view_model
 
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import ru.kryu.playlistmaker.search.ui.models.TrackForUi
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.launch
+import ru.kryu.playlistmaker.favourite.domain.FavouritesInteractor
+import ru.kryu.playlistmaker.search.domain.model.Track
+import ru.kryu.playlistmaker.search.ui.mapper.TrackToTrackForUi
 
-class FavouritesViewModel : ViewModel() {
+class FavouritesViewModel(
+    private val favouritesInteractor: FavouritesInteractor,
+    private val trackToTrackForUi: TrackToTrackForUi,
+) : ViewModel() {
 
-    private val listFavouritesMutableLiveData = MutableLiveData<List<TrackForUi>>()
-    val listFavouritesLiveData = listFavouritesMutableLiveData
+    private val favouritesMutableLiveData = MutableLiveData<FavouritesState>()
+    val favouritesLiveData = favouritesMutableLiveData
```

Comment:
Тут получается та же самая MutableLiveData наружу отдаётся, нужно так:
```suggestion
    val favouritesLiveData: LiveData<FavouritesState> = favouritesMutableLiveData
```